$NetBSD$

Changes:

1) Make libm, libz, libjansson and libiconv dynamically linked, instead of statically.
2) Use system libjansson instead of the one included with zero-epwing.
3) Link libintl if |UNIX AND NOT APPLE|.
4) Specify the full path to the libeb static library, since (at least on macOS) the linker
   has trouble linking it otherwise.

--- CMakeLists.txt.orig
+++ CMakeLists.txt
@@ -15,11 +15,20 @@
 option(JANSSON_BUILD_DOCS "" OFF)
 option(JANSSON_WITHOUT_TESTS "" ON)
 add_subdirectory(jansson)
-link_directories(eb/eb/.libs ${CMAKE_BINARY_DIR}/jansson/lib)
+link_directories(eb/eb/.libs)
 add_executable(zero-epwing main.c book.c convert.c hooks.c)
 add_dependencies(zero-epwing eb jansson)
-target_link_libraries(zero-epwing libeb.a libz.a libjansson.a)
+target_link_libraries(zero-epwing
+    z jansson
+    ${CMAKE_CURRENT_SOURCE_DIR}/eb/eb/.libs/libeb.a
+)
+
+# Dynamically link libintl on non-Apple UNIX-alikes
+if (UNIX AND NOT APPLE)
+    target_link_libraries(zero-epwing intl)
+endif (UNIX AND NOT APPLE)
+
+# Dynamically link libiconv on Windows or macOS
 if (WIN32 OR APPLE)
-    target_link_libraries(zero-epwing libiconv.a)
+    target_link_libraries(zero-epwing iconv)
 endif (WIN32 OR APPLE)
-target_link_libraries(zero-epwing)
